src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -659,22 +659,11 @@ public sealed override string Name
 
                     if (IsExplicitInterfaceImplementation)
                     {
-                        PropertySymbol? explicitlyImplementedPropertyOpt = _property.ExplicitInterfaceImplementations.FirstOrDefault();
+                        string accessorName = GetAccessorName(_property.Name,
+                                isGetMethod, isWinMdOutput: _property.IsCompilationOutputWinMdObj()); //Not name - could be indexer placeholder
 
-                        if (explicitlyImplementedPropertyOpt is object)
-                        {
-                            MethodSymbol? implementedAccessor = isGetMethod
-                                ? explicitlyImplementedPropertyOpt.GetMethod
-                                : explicitlyImplementedPropertyOpt.SetMethod;
-
-                            string accessorName = (object)implementedAccessor != null
-                                ? implementedAccessor.Name
-                                : GetAccessorName(explicitlyImplementedPropertyOpt.MetadataName,
-                                    isGetMethod, isWinMdOutput: _property.IsCompilationOutputWinMdObj()); //Not name - could be indexer placeholder
-
-                            string? aliasQualifierOpt = _property.GetExplicitInterfaceSpecifier()?.Name.GetAliasQualifierOpt();
-                            name = ExplicitInterfaceHelpers.GetMemberName(accessorName, explicitlyImplementedPropertyOpt.ContainingType, aliasQualifierOpt);
-                        }
+                        var explicitInterfaceSpecifier = _property.GetExplicitInterfaceSpecifier();
+                        name = ExplicitInterfaceHelpers.GetMemberName(explicitInterfaceSpecifier, accessorName);
```

Comment:
>name = ExplicitInterfaceHelpers.GetMemberName(explicitInterfaceSpecifier, accessorName); [](http://example.com/codeflow?start=24&length=88)

For an accessor, we probably want MetadataName to be based on the name of the implemented accessor.